ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/StandaloneExecutorTest.java

```diff
@@ -16,78 +16,288 @@
-        .andReturn(Collections.singletonList(query));
-    query.start();
-    EasyMock.expectLastCall();
-    EasyMock.replay(query, engine);
+  public void shouldRunUnSetStatements() throws IOException {
 
-    executor.start();
+    final StandaloneExecutor standaloneExecutor =
+        new StandaloneExecutor(ksqlConfig, engine, queriesFile, udfLoader);
+
+    final List<Pair<String, Statement>> statementPairs = new ArrayList<>();
+    final MetaStore metaStore = EasyMock.niceMock(MetaStore.class);
+
+    EasyMock.expect(engine.parseStatements(anyString(), anyObject())).andReturn(statementPairs);
+    EasyMock.expect(engine.getMetaStore()).andReturn(metaStore);
+
+    final SetProperty setProperty = new SetProperty(Optional.empty(), "name", "value");
+    statementPairs.add(new Pair<>("SET", setProperty));
+
+    final UnsetProperty unsetProperty = new UnsetProperty(Optional.empty(), "name");
+    statementPairs.add(new Pair<>("UNSET", unsetProperty));
+
+    EasyMock.replay(engine);
+    standaloneExecutor.start();
+    EasyMock.verify(engine);
+
+    assertTrue(standaloneExecutor.getConfigProperties().isEmpty());
 
-    EasyMock.verify(query);
   }
 
   @Test
-  public void shouldNotExecuteNonPersistentQueries() {
-    final QueryMetadata query = EasyMock.createMock(QueryMetadata.class);
-    EasyMock.expect(engine.createQueries(anyString(), eq(ksqlConfig)))
-        .andReturn(Collections.singletonList(query));
-    EasyMock.expect(query.getStatementString()).andReturn("");
-    EasyMock.replay(query, engine);
+  @SuppressWarnings("unchecked")
+  public void shouldRunCsasStatements() throws IOException {
+
+    final StandaloneExecutor standaloneExecutor =
+        new StandaloneExecutor(ksqlConfig, engine, queriesFile, udfLoader);
+
+    final List<Pair<String, Statement>> statementPairs = new ArrayList<>();
+    final MetaStore metaStore = EasyMock.niceMock(MetaStore.class);
 
-    executor.start();
+    final Query csas1Query = EasyMock.niceMock(Query.class);
+    final CreateStreamAsSelect createStreamAsSelect1 = new CreateStreamAsSelect(EasyMock.niceMock(QualifiedName.class), csas1Query, false, Collections.emptyMap(), Optional.empty());
+    statementPairs.add(new Pair<>("CSAS1", createStreamAsSelect1));
+
+    EasyMock.expect(engine.parseStatements(anyString(), anyObject())).andReturn(statementPairs);
+    EasyMock.expect(engine.getMetaStore()).andReturn(metaStore);
+
+    final QueryMetadata csas1QueryMetadata = EasyMock.niceMock(PersistentQueryMetadata.class);
+    EasyMock.expect(csas1QueryMetadata.getDataSourceType()).andReturn(DataSourceType.KSTREAM);
+    EasyMock.expect(engine.buildMultipleQueries(eq("CSAS1"), anyObject(KsqlConfig.class), anyObject(Map.class)))
+        .andReturn(Collections.singletonList(csas1QueryMetadata)).once();
+    EasyMock.expect(engine.getQueryExecutionPlan(csas1Query, ksqlConfig))
+        .andReturn(csas1QueryMetadata).once();
+
+    EasyMock.replay(csas1Query);
```

Comment:
`EastMock.replay` and `verify` take varags, so you can collapse these multiple lines into one:

```
EasyMock.replay(v1, v2, v3);
start();
EasyMock.verify(v1, v2, v3);
```